enterprise/server/backends/pebble_cache/pebble_cache.go

```diff
@@ -401,6 +406,105 @@ func olderThanThreshold(t time.Time, threshold time.Duration) bool {
 	return age >= threshold
 }
 
+// databaseVersionKey returns the key bytes of a key where a serialized,
+// database-wide version metadata proto is stored.
+func (p *PebbleCache) databaseVersionKey() []byte {
+	var key []byte
+	key = append(key, SystemKeyPrefix...)
+	key = append(key, []byte("database-version")...)
+	return key
+}
+
+// databaseVersionKey returns the database-wide version metadata which
+// contains the database version.
+func (p *PebbleCache) databaseVersionMetadata() (*rfpb.VersionMetadata, error) {
+	db, err := p.leaser.DB()
+	if err != nil {
+		return nil, err
+	}
+	defer db.Close()
+
+	buf, err := pebbleutil.GetCopy(db, p.databaseVersionKey())
+	if err != nil {
+		if status.IsNotFoundError(err) {
+			// If the key is not present in the DB; return an empty
+			// proto.
+			return &rfpb.VersionMetadata{}, nil
+		}
+		return nil, err
+	}
+
+	versionMetadata := &rfpb.VersionMetadata{}
+	if err := proto.Unmarshal(buf, versionMetadata); err != nil {
+		return nil, err
+	}
+	return versionMetadata, nil
+}
+
+// currentDatabaseVersion returns the currently stored filestore.PebbleKeyVersion.
+// It is safe to call this function in a loop -- the underlying metadata will
+// only be fetched a max of once per second.
+func (p *PebbleCache) currentDatabaseVersion() (filestore.PebbleKeyVersion, error) {
+	unlockFn := p.locker.RLock(string(p.databaseVersionKey()))
+
+	if time.Since(p.lastDBVersionUpdate) > time.Second {
```

Comment:
Is it not sufficient to read the value once on startup and then rely on the current value is in p.lastDBVersion? Under what circumstances would the database version change under us in the DB?

Reply:
Done.